src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -91,32 +117,45 @@ public virtual FunctionSecrets GetFunctionSecrets(string functionName)
 
             functionName = functionName.ToLowerInvariant();
 
-            return _secretsMap.GetOrAdd(functionName, (n) =>
+            return _secretsMap.GetOrAdd(functionName, n =>
             {
-                FunctionSecrets secrets;
+                IList<Key> secrets;
                 string secretFileName = string.Format(CultureInfo.InvariantCulture, "{0}.json", functionName);
                 string secretFilePath = Path.Combine(_secretsPath, secretFileName);
                 if (File.Exists(secretFilePath))
                 {
                     // load the secrets file
                     string secretsJson = File.ReadAllText(secretFilePath);
-                    secrets = JsonConvert.DeserializeObject<FunctionSecrets>(secretsJson);
+                    secrets = ScriptSecretSerializer.DeserializeFunctionSecrets(secretsJson);
                 }
                 else
                 {
-                    // initialize with new secrets and save it
-                    secrets = new FunctionSecrets
+                    // initialize with new list with a default secret and save it
+                    secrets = new List<Key>
                     {
-                        Key = GenerateSecretString()
+                        GenerateSecret(DefaultFunctionKeyName)
                     };
 
-                    File.WriteAllText(secretFilePath, JsonConvert.SerializeObject(secrets, Formatting.Indented));
+                    string secretsContent = ScriptSecretSerializer.SerializeFunctionSecrets(secrets);
+                    File.WriteAllText(secretFilePath, secretsContent);
                 }
 
-                return secrets;
+                return secrets.ToDictionary(s => s.Name, s => ReadSecretValue(s));
             });
```

Comment:
I may be missing something, but the key itself is already an object (`Key`) and we just store the information as a collection/array of them. If we were to introduce a different type here, we'd have a simple wrapper that would expose a list of keys (or the dictionary we're currently returning, unless we want to introduce a new type to avoid returning irrelevant information) and a version. 
The way the content is actually serialized is independent of this code.